iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_sender.hpp

```diff
@@ -92,9 +93,21 @@ class ChunkSender : public ChunkDistributor<typename ChunkSenderDataType::ChunkD
     void release(const mepoo::ChunkHeader* const chunkHeader) noexcept;
 
     /// @brief Send an allocated chunk to all connected ChunkQueuePopper
-    /// @param[in] chunkHeader, pointer to the ChunkHeader to send
+    /// @param[in] chunkHeader, pointer to the ChunkHeader to send; the ownership of the pointer is transferred to this
+    /// method
     void send(mepoo::ChunkHeader* const chunkHeader) noexcept;
 
+    /// @brief Send an allocated chunk to a specific ChunkQueuePopper
+    /// @param[in] chunkHeader, pointer to the ChunkHeader to send; the ownership of the pointer is transferred to this
+    /// method
+    /// @param[in] uniqueQueueId is an unique ID which identifies the queue to which this chunk shall be delivered
+    /// @param[in] lastKnownQueueIndex is used for a fast lookup of the queue with uniqueQueueId
+    /// @return true when successful, false otherwise
+    /// @note This method does not add the chunk to the history
+    bool sendToQueue(mepoo::ChunkHeader* const chunkHeader,
```

Comment:
Is it valid to have the `nullptr` as argument for `chunkHeader`? If not I would change the `chunkHeader` argument into a reference to communicate this to the user that this is never a `nullptr`.

The same argument goes also for the other methods but you did not touch the fishy, but if you touch the fishy you can turn the pointer into a reference.

https://www.youtube.com/watch?v=8gdsueMGUfk

Reply:
Well, it is the same behavior as with `ChunkSender::send`. One can argue if that is the desired behavior but since that can easily becomes a bottomless pit I would argue it should be discussed/implemented in a separate issue.